model/plugin_cluster_event.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+const (
+	PluginClusterEventSendTypeReliable   = CLUSTER_SEND_RELIABLE
+	PluginClusterEventSendTypeBestEffort = CLUSTER_SEND_BEST_EFFORT
+)
+
+// PluginClusterEvent is used to allow intra-cluster plugin communication.
+type PluginClusterEvent struct {
+	// Id is the unique identifier for the event.
+	Id string
+	// Data is the event payload.
+	Data []byte
+}
+
+// PluginClusterEventSendOptions defines some properties that apply when sending
+// plugin events across a cluster.
+type PluginClusterEventSendOptions struct {
+	// SendType defines the type of communication channel used to send the event.
+	SendType string
+	// ReceiverId identifies which node the message should be sent to.
```

Comment:
I feel sender/receiver makes it slightly confusing because "receiver" can mean from both directions. source/destination makes it slightly more clear. 0/5.

Reply:
I see, but this is an object defining sending options so I assumed the receiver could only be the one we are sending the event to. No issue with changing the naming here but I am not sure how this differs from source/destination. They can be both sources/destinations same way they can be both senders/receivers but it's the context that removes the ambiguity.

Maybe we could just name it `TargetId` to get us out of these semantics.

Reply:
Yeah, destination could very well work both ways. It's just how I perceived the meaning. No strong reasons.